pm: Make device pm optional per state - V2 (#70623)
zephyrproject-rtos/zephyr: doc/services/pm/device.rst

System-Managed Device Power Management
======================================

The system managed device power management (PM) framework is a method where
devices are suspended along with the system entering a CPU (or SoC) power state.
It can be enabled by setting :kconfig:option:`CONFIG_PM_DEVICE_SYSTEM_MANAGED`.
When using this method, device power management is mostly done inside
:c:func:`pm_system_suspend()`.

If a decision to enter a CPU lower power state is made, the power management
subsystem will check if the selected low power state triggers device power
management and then suspend devices before changing state. The subsystem takes
care of suspending devices following their initialization order, ensuring that
possible dependencies between them are satisfied. As soon as the CPU wakes up
from a sleep state, devices are resumed in the opposite order that they were
suspended.

> The decision about suspending devices in a low power state is done based on the
> state and if it has setting the property ``zephyr,pm-device-disabled``. Here is

Reviewer: ```suggestion
The decision about suspending devices when entering a low power state is done based on the
state and if it has set the property ``zephyr,pm-device-disabled``. Here is
```

> an example of a target with two low power states with only triggering device power
> management:

Reviewer: ```suggestion
an example of a target with two low power states with only one triggering device power
management:
```
